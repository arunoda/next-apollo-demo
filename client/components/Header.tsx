import React from 'react';
import styled from 'styled-components';
import Link from 'next/link'

const links = [
    { label: 'Home', href: '/' },
    { label: 'About', href: '/about' },
    { label: 'Contacts', href: '/contacts' }
]
    
const Header = () => {
    return (
        <HeaderContainer>
            <HeaderBrand>
                Next Apollo Demo
            </HeaderBrand>
            <HeaderLinks>
                {
                    links.map((link, id) => (
                        <Link href={link.href} key={id}><a>{link.label}</a></Link>
                    ))
                }
            </HeaderLinks>
        </HeaderContainer>
    )
}

const HeaderContainer = styled.div`
    display:flex;
    position:sticky;
    width:auto;
    color:black;
    border-bottom:1px solid black;
    height:40px;
    align-items:center;
    justify-content:space-between;
    padding: 1rem 1rem;
    z-index: 50;
`
const HeaderBrand = styled.h2`
 @media (max-width: 650px) {
       font-size:10px;
    }
`;

const HeaderLinks = styled.div`
     display: flex;
    list-style: none;
    align-items: center;
    justify-content: flex-end;
    > a {
        margin: 1.5rem;
        text-decoration: none;
        color: #000;
        display: flex;
        align-items: center;
        font-size: 1rem;
        cursor: pointer;
        &:hover{
          color:#3f51b5;
        }
        &:active{
            color:#3f51b5;
        }
    }
`

export default Header;