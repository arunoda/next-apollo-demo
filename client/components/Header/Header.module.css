.header {
  color: var(--grey);
  background: var(--white);
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 100;
}

.wrapper {
  max-width: var(--maxWidth);
  margin: 0 auto;
  width: 100%;
}

.nav {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  min-height: 40px;
}

.navbar {
  display: flex;
  list-style: none;
}

.navItem {
  padding: 20px 0;
  text-transform: uppercase;
  min-width: 100px;
}