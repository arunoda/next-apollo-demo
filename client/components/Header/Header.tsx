import Link from 'next/link';
import styles from './Header.module.css';

export default function Navigation() {
  return (
    <header role="banner" aria-label="Next-Apollo" className={styles.header}>
      <nav role="menubar" className={styles.nav}>
        <div className={styles.wrapper}>
          <ul role="menubar" className={styles.navbar}>
            <li role="menuitem" className={styles.navItem}>
              <Link href="/">Home</Link>
            </li>
            <li role="menuitem" className={styles.navItem}>
              <Link href="/users">Users</Link>
            </li>
          </ul>
        </div>
      </nav>
    </header>
  );
}
