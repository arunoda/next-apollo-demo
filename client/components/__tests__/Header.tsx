import Header from '../Header';
import { cleanup,render } from '@testing-library/react';

describe("Header Component",()=>{
        afterEach(cleanup);
        describe("display Header",()=>{
            test("it should match snapshot",async()=>{
                const {container}= render(
                    <Header />
                    )
                    expect(container).toMatchSnapshot();
                    const text = container.getElementsByTagName('h2')[0].textContent;
                    expect(text).toBe('Next Apollo Demo')
            })
        })
})