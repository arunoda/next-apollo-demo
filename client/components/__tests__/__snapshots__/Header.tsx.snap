// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component display Header it should match snapshot 1`] = `
<div>
  <div
    class="Header__HeaderContainer-sc-150phg5-0 fDQNNS"
  >
    <h2
      class="Header__HeaderBrand-sc-150phg5-1 eHzWZO"
    >
      Next Apollo Demo
    </h2>
    <div
      class="Header__HeaderLinks-sc-150phg5-2 pKijZ"
    >
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/about"
      >
        About
      </a>
      <a
        href="/contacts"
      >
        Contacts
      </a>
    </div>
  </div>
</div>
`;
